Backendless/Push/Registrar.cs

```diff
@@ -1,49 +1,81 @@
+      deviceRegistration.DeviceId = Backendless.Messaging.DeviceID;
+      deviceRegistration.Channels = channels;
+      
+      if( Environment.OSVersion.Platform.ToString() == "Unix" )
+        deviceRegistration.Os = "ANDROID";
+      else
+        deviceRegistration.Os = "IOS";
 
-    private static EventHandler<NotificationChannelConnectionEventArgs> _onConnectionStatusChangedHandler;
-    private static EventHandler<NotificationChannelUriEventArgs> _channelUriUpdateddHandler;
+      return Invoker.InvokeSync<String>( MessagingService.DEVICE_REGISTRATION_MANAGER_SERVER_ALIAS, "registerDevice", new Object[] { deviceRegistration } );  
+    }
 
-    internal static void RegisterDevice( String channel, PushNotificationsBinding pushNotificationsBinding,
-                                         AsyncCallback<String> callback )
+    internal static Boolean UnregisterDevice()
     {
-      if( String.IsNullOrEmpty( channel ) )
-        throw new ArgumentNullException( "Push channel cannot be null" );
-
-      if( _currentRegistration == null || !_currentRegistration.IsRegistered() )
-        MakeInternalRegistration( channel, pushNotificationsBinding, callback );
-      else
-        callback.ResponseHandler.Invoke( _currentRegistration.GetRegistrationId() );
+      return Invoker.InvokeSync<Boolean>( MessagingService.DEVICE_REGISTRATION_MANAGER_SERVER_ALIAS, "unregisterDevice", new object[] { Backendless.Messaging.DeviceID } );
     }
 
-    private static void MakeInternalRegistration( String channel, PushNotificationsBinding pushNotificationsBinding, AsyncCallback<String> callback )
+    internal static Int32? UnregisterDevice( IList<String> channels )
     {
-      var httpNotificationChannel = HttpNotificationChannel.Find( channel );
+      return Invoker.InvokeSync<Int32?>( MessagingService.DEVICE_REGISTRATION_MANAGER_SERVER_ALIAS, "unregisterDevice", new Object[] { Backendless.Messaging.DeviceID, channels } );
+    }
+  }
+}
 
-      if( httpNotificationChannel == null )
-      {
-          httpNotificationChannel = new HttpNotificationChannel( channel, Backendless.URL );
-          httpNotificationChannel.ChannelUriUpdated += ( sender, args ) => ProceedRegistration( httpNotificationChannel, callback );
-          httpNotificationChannel.Open();
-      }
-      else
-      {
-          httpNotificationChannel.ChannelUriUpdated += ( sender, args ) => ProceedRegistration( httpNotificationChannel, callback );
-      }
+/*
+private static RegistrationDecorator _currentRegistration;
 
-         if(pushNotificationsBinding != null)
-            pushNotificationsBinding.ApplyTo( httpNotificationChannel );
-                                 
+private static EventHandler<NotificationChannelConnectionEventArgs> _onConnectionStatusChangedHandler;
+private static EventHandler<NotificationChannelUriEventArgs> _channelUriUpdateddHandler;
 
+internal static void RegisterDevice( String channel, PushNotificationsBinding pushNotificationsBinding,

```

Comment:
Is there RegisterDeviceAsync method?

Reply:
Yes it is.
https://github.com/Backendless/.NET-SDK/pull/33/commits/a63f128b259208463cc3ad183000e970002e07f7#diff-86a43d5752b84f2977969cff2e53d043R103